core/src/main/java/hudson/model/Queue.java

```diff
@@ -358,7 +358,6 @@ public LoadBalancer getLoadBalancer() {
     }
 
     public void setLoadBalancer(@NonNull LoadBalancer loadBalancer) {
-        if(loadBalancer==null)  throw new IllegalArgumentException();
```

Comment:
I checked the ecosystem, everytime this method is called with a valid nonNull loadBalancer.